hazelcast/src/main/java/com/hazelcast/spi/Operation.java

```diff
@@ -571,15 +571,20 @@ public final void readData(ObjectDataInput in) throws IOException {
 
     protected abstract void readInternal(ObjectDataInput in) throws IOException;
 
+    protected void toString(StringBuilder sb) {
+    }
+
```

Comment:
I like the abstract method, since you cannot forget about the implementation. You anyway did it for most of the classes.

And what convention should be needed? `Operation` is abstract so it's fine to have `public abstract String toString(StringBuilder sb);` there. Then we make `public final String toString()` so it cannot be overridden by accident and use the abstract method in there. What else should be needed?